contracts/RewardManagerFactory.sol

```diff
@@ -0,0 +1,216 @@
+                endDistribution: _endDistribution
+            })
+        ); //stacking up every crowdsale info ever made to crowdsales variable
+
+        emit RewardManagerLaunched(
+            address(newManager),
+            _startDistribution,
+            _endDistribution
+        );
+        totalRewardManagers++;
+    }
+
+    function removeRewardManager(uint256 _index) public onlyOwner {
+        require(_index <= totalRewardManagers, "Invalid Index");
+        delete managers[_index];
+    }
+
+    function userTotalVestingInfo(address _user)
+        public
+        view
+        returns (
+            uint256 totalVested,
+            uint256 totalDrawnAmount,
+            uint256 amountBurnt,
+            uint256 claimable,
+            uint256 bonusRewards,
+            uint256 stillDue
+        )
+    {
+        UserInfo memory user;
+        for (uint256 i = 0; i < totalRewardManagers; i++) {
+            address rewardManagerAddress = managers[i].managerAddress;
+            if (rewardManagerAddress != address(0)) {
+                RewardManager manager = RewardManager(rewardManagerAddress);
+                (
+                    user._totalVested,
+                    user._totalDrawnAmount,
+                    user._amountBurnt,
+                    user._claimable,
+                    user._bonusRewards,
+                    user._stillDue
+                ) = manager.vestingInfo(_user);
+                
+                if(user._totalVested) {
+                    totalVested += user._totalVested;
+                    totalDrawnAmount += user._totalDrawnAmount;
+                    amountBurnt += user._amountBurnt;
+                    claimable += user._claimable;
+                    bonusRewards += user._bonusRewards;
+                    stillDue += user._stillDue;
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice Draws down any vested tokens due in all Reward Manager
+     * @dev Must be called directly by the beneficiary assigned the tokens in the vesting
+     */
+    function drawDown() external onlyOwner {
+        for (uint256 i = 0; i < totalRewardManagers; i++) {
+            address rewardManagerAddress = managers[i].managerAddress;
+            if (rewardManagerAddress != address(0)) {
+                RewardManager manager = RewardManager(rewardManagerAddress);
+                manager.drawDown(msg.sender);
+            }
+        }
+    }
+
+    /**
+     * @notice Pre maturely Draws down all vested tokens by burning the preMaturePenalty
+     * @dev Must be called directly by the beneficiary assigned the tokens in the vesting
+     */
+    function preMatureDraw() external onlyOwner {
+        for (uint256 i = 0; i < totalRewardManagers; i++) {
+            address rewardManagerAddress = managers[i].managerAddress;
+            if (rewardManagerAddress != address(0)) {
+                RewardManager manager = RewardManager(rewardManagerAddress);
```

Comment:
`if ( _totalVested.sub(_totalDrawnAmount) > 0`

Reply:
but this logic is being already handled in preMatureDraw in RewardManager & any user who's remaining Balance equals to 0 will receive error while triggering this function in factory